app/celery/tasks.py

```diff
@@ -84,3 +90,62 @@ def copy_redaction_failed_pdf(source_filename):
     # e.g. if a file has 5 days left to expire on a ONE_WEEK retention in the source bucket,
     # in the destination bucket the expiration time will be reset to 7 days left to expire
     obj.copy(copy_source, ExtraArgs={'ServerSideEncryption': 'AES256'})
+
+
+@notify_celery.task(name='create-letter-pdf')
+@statsd(namespace="template_preview")
+def create_letter_pdf(encrypted_letter_data):
+    letter_details = current_app.encryption_client.decrypt(encrypted_letter_data)
+    current_app.logger.info('Creating a pdf for notification with id {}'.format(letter_details["notification_id"]))
+    logo_filename = f'{letter_details["logo_filename"]}.svg' if letter_details['logo_filename'] else None
+
+    template = LetterPrintTemplate(
+        letter_details['template'],
+        values=letter_details['values'] or None,
+        contact_block=letter_details['letter_contact_block'],
+        # letter assets are hosted on s3
+        admin_base_url=current_app.config['LETTER_LOGO_URL'],
+        logo_file_name=logo_filename,
+    )
+    with current_app.test_request_context(''):
```

Comment:
why do we need to use the test_request_context here?

Reply:
because instead of HTTP endpoint, our function is now a Celery task, and the HTML function it calls needs a request context that celery does not provide.

Or in other words, HTML function started throwing an exception about being outside of request context when I moved it onto a task, and I looked in weasyprint docs and found this little hack that made it work 😅